### PR TITLE
Split out mock propolis-server from real thing

### LINKED ISSUE #553: Split mock-server from propolis-server
The mock implementation of propolis-server is built by setting a feature flag which elides _most_ of the actually-run-a-VM code, but it still includes bits from propolis-lib for simulated the UART.  This leads consumers of the mock server (including simulated sled-agent in Omicron) to depend on practically all of crates required for "real" propolis.  This is unfortunate for several reasons:
1. They must build more software than is strictly necessary which...
a. Requires dependencies of Propolis (such as Crucible) to be kept in lockstep with what's present in Omicron
2. It introduces a pseudo-circular dependency between Omicron and Propolis

We should split out the mock server from propolis-server, and reimplement its UART bits so that it no longer depends on propolis-lib, or much beyond the usual suspects (like Dropshot, serde, etc).

### PR DESCRIPTION
**Pull request by pfmooney**
The simulated sled-agent in Omicron uses the Propolis mock server as a stand-in for a real VM instance.  In order to break an awkward dependencies situation, it would be nice if the mock server did not haul in all everything which propolis-server requires.

Fixes #553

### CODE REVIEW

**Review by gjcolombo — APPROVED** [review]
LGTM with one small tweak. Thanks for the changes!

File: bin/mock-server/src/lib/copied.rs
[the remark this answers is not in the document]
Comment (pfmooney) [reply bin/mock-server/src/lib/copied.rs:1]: If you're alright with that change being slightly delayed, my [follow-on](https://github.com/pfmooney/propolis/commit/49191153513fff8be5b49ae7b1b28408caff3233) for this does clean up that error case.

[the remark this answers is not in the document]
Comment (gjcolombo) [reply bin/mock-server/src/lib/copied.rs:1]: I think I'm suggesting something more radical than what's in 4919115: not only can we drop the `InnerBuilderError` variant from `mock_server::copied::ServerSpecBuilderError`, I suspect we can get rid of the entire enum and just have `mock_server::copied::slot_to_pci_path` return some other error type for its `Err` variant (at which point I'd probably just get rid of the `copied` module entirely and just move the private copy of `slot_to_pci_path` to the lib).

All that said, this is just cleanup and aesthetics, no need to hold this particular PR for it.

[the remark this answers is not in the document]
Comment (pfmooney) [reply bin/mock-server/src/lib/copied.rs:1]: If you're happy with clean-up like that happening later, then I'm inclined to move forward with things as-is.

[the remark this answers is not in the document]
Comment (gjcolombo) [reply bin/mock-server/src/lib/copied.rs:1]: Yep, we can tackle that in a later PR.

**Pull request merged**